homeassistant/components/ecobee/weather.py

```diff
@@ -61,6 +61,12 @@ def name(self):
         """Return the name of the sensor."""
         return self._name
 
+    @property
+    def unique_id(self):
+        """Return a unique identifier for the weather platform."""
+        thermostat = self.data.ecobee.get_thermostat(self._index)
+        return f"{thermostat['identifier']}-weather"
```

Comment:
The entity registry is aware of both the integration domain and the platform domain. So we shouldn't need to add `"weather"`.

Reply:
So in other words the unique_id only has to be unique within the weather domain?

Reply:
Within the weather domain, within the ecobee integration.